widgetssdk/src/main/java/com/glia/widgets/call/domain/HandleCallPermissionsUseCase.kt

```diff
@@ -0,0 +1,27 @@
+package com.glia.widgets.call.domain
+
+import com.glia.androidsdk.Engagement
+import com.glia.widgets.permissions.PermissionsGrantedCallback
+import com.glia.widgets.permissions.domain.PermissionsForEngagementMediaTypeUseCase
+import com.glia.widgets.permissions.domain.PermissionsHandlerUseCase
+
+internal class HandleCallPermissionsUseCase(
+    private val permissionsForEngagementMediaTypeUseCase: PermissionsForEngagementMediaTypeUseCase,
+    private val permissionsHandlerUseCase: PermissionsHandlerUseCase
+) {
+    operator fun invoke(isUpgradeToCall: Boolean,
+                        mediaType: Engagement.MediaType,
+                        callback: PermissionsGrantedCallback) {
+        if (isUpgradeToCall) {
+            callback(true)
+            return
+        }
```

Comment:
Why do we even need to execute `HandleCallPermissionsUseCase` for `isUpgradeToCall` if permissions are not needed for this case?

Reply:
Yes, it should be handled outside the UseCase.